sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerFactory.java

```diff
@@ -57,36 +61,52 @@
       .build();
 
   private final DruidSchema druidSchema;
+  private final TimelineServerView serverView;
   private final QueryLifecycleFactory queryLifecycleFactory;
   private final DruidOperatorTable operatorTable;
   private final ExprMacroTable macroTable;
   private final PlannerConfig plannerConfig;
   private final ObjectMapper jsonMapper;
   private final AuthorizerMapper authorizerMapper;
+  private final DruidLeaderClient coordinatorDruidLeaderClient;
+  private final DruidLeaderClient overlordDruidLeaderClient;
 
   @Inject
   public PlannerFactory(
       final DruidSchema druidSchema,
+      final TimelineServerView serverView,
       final QueryLifecycleFactory queryLifecycleFactory,
       final DruidOperatorTable operatorTable,
       final ExprMacroTable macroTable,
       final PlannerConfig plannerConfig,
       final AuthorizerMapper authorizerMapper,
-      final @Json ObjectMapper jsonMapper
+      final @Json ObjectMapper jsonMapper,
+      final @Coordinator DruidLeaderClient coordinatorDruidLeaderClient,
+      final @IndexingService DruidLeaderClient overlordDruidLeaderClient
```

Comment:
Regarding my comment about mocking `DruidLeaderClient` in benchmark, I think the real issue is that `PlannerFactory` needs `TimelineServerView`, `ObjectMapper`, and `DruidLeaderClient`s. Since this class is just a factory which is responsible for creating a new planner, it's not supposed to get information from TimelineServerview or leaderClients on its own. This makes me confused until tracking the whole change related to those parameters. 

I think the same design for `DruidSchema` can be applied to here too. `DruidSchema` also needs `TimelineServerView`, but it's being injected and so `TimelineServerView` is not exposed here. Similarly, you can inject `SystemSchema` instead of these parameters. Actually, you've already made it injectable. Why don't you use it?

Reply:
Thanks.